### DISCUSSION TITLE
Diamond Template: Collapsed sidebar opens/closes on reload

### DISCUSSION THREAD

**Discussion by f1nnk** [primefaces/community#2741] [PrimeFaces Templates]
When the sidebar/menu is collapsed and the page gets reloaded, the sidebar briefly opens and closes again.
You can replicate this in the [Showcase](http://www.primefaces.org/diamond/panel.xhtml) by collapsing the sidebar and then
either reload the page or press the "Login" button further down on the page. The same thing happens when the sidebar is empty (has no content) and therefore is already collapsed.
The expected behaviour would be that it remains closed without any animation when it has no content or is collapsed. Is that possible?

**Comment by melloware** [comment]
I agree I can replicate your issue. let me see if I can hack it.

**Reply by f1nnk** [reply]
@melloware is there any update on this issue? did you succeed? 😄

**Reply by melloware** [reply]
I was not successful in the time i spent on it. It seems baked into the layout.js code that comes with the theme and after 10 minutes I realized it wouldn't be a simple change.

**Reply by f1nnk** [reply]
I looked into it a little bit and found out that the class `layout-static-inactive` to collapse the sidebar gets set in the `restoreMenuState` method.
```javascript
var staticMenuState = localStorage.getItem('diamond_static_menu_inactive');
if (staticMenuState) {
    this.wrapper.addClass('layout-static-inactive layout-static-inactive-restore');
}
```
I assume the sidebar briefly expands on page reload before collapsing due to the delay in applying the class after rendering. I guess the only option is to fetch the item from localStorage and apply the class before the page loads. Do you have an idea?

**Reply by melloware** [reply]
Yep I found the same thing @f1nnk but since this is done with JS code i couldn't see a way of doing it without there being some flicker.  At least not obvious since it loads one way and then basically "switches" to the localStorage value.
